### PR TITLE
[XRay Sampler] Fix - Ensure all XRay Sampler functionality is under ParentBased logic

### PR DESCRIPTION
**Pull request by jj22ee**
### Issue #, if available:

This PR fixes https://github.com/aws-observability/aws-otel-js-instrumentation/issues/79
This fixes the bug where:
- While the subcomponents of XRay Sampler uses ParentBased logic to short-circuit the Parent's Sampling Decision, the XRay Sampling Statistics recording logic is not skipped when a Parent's Sampling Decision is found. This causes XRay Sampler to produce Sampling Statistics based on number of Spans (regardless of Parent's Sampling Decision), while XRay Sampler should produce Sampling Statistics based on number of Traces (aka the number of root Spans that makes the Sampling Decision)

###  Description of changes:
1. Wrap entire XRay Sampler Internal Logic under a single ParentBased Sampler
2. Remove use of redundant ParentBased Sampler logic for internal subcomponents for XRay Sampler.

### Testing:
1. Added unit tests to verify ParentBased sampling decisions
3. Original Sampler functionality tested locally using [Sampler Test Bed](https://github.com/aws-observability/aws-otel-community/tree/master/centralized-sampling-tests)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-distro-opentelemetry-node-autoinstrumentation/test/sampler/aws-xray-remote-sampler.test.ts [hunk 1]
```diff
@@ -187,16 +186,66 @@ describe('AwsXrayRemoteSampler', () => {
...
+
+    const sampler: _AwsXRayRemoteSampler = new _AwsXRayRemoteSampler({ resource: Resource.EMPTY });
+    const tracerProvider: NodeTracerProvider = new NodeTracerProvider({ sampler: sampler });
+    const tracer: Tracer = tracerProvider.getTracer('test');
+
+    setTimeout(() => {
+      tracer.startActiveSpan('test0', span0 => {
+        const span1: Span = tracer.startSpan('test1');
+        const span2: Span = tracer.startSpan('test2');
+        span2.end();
```
Comment (pxaws) [inline aws-distro-opentelemetry-node-autoinstrumentation/test/sampler/aws-xray-remote-sampler.test.ts:242]: Could you explain why span 1 and 2 are not children of span 0? Maybe add some comments here?

Comment (jj22ee) [reply aws-distro-opentelemetry-node-autoinstrumentation/test/sampler/aws-xray-remote-sampler.test.ts:242]: In both tests, span 1 and 2 are both children of span 0. Will add comments to clarify.

File: aws-distro-opentelemetry-node-autoinstrumentation/src/sampler/aws-xray-remote-sampler.ts [hunk 2]
```diff
@@ -22,12 +22,35 @@ const DEFAULT_RULES_POLLING_INTERVAL_SECONDS: number = 5 * 60;
...
+  ): SamplingResult {
+    return this._root.shouldSample(context, traceId, spanName, spanKind, attributes, links);
+  }
+
+  public toString(): string {
+    return `AwsXRayRemoteSampler{root=${this._root.toString()}`;
+  }
+}
+
+export class _AwsXRayRemoteSampler implements Sampler {
```
[the remark this answers is not in the document]
Comment (jj22ee) [reply aws-distro-opentelemetry-node-autoinstrumentation/src/sampler/aws-xray-remote-sampler.ts:48]: If we want to use it in tests, we need to have the `export`. However, as long as this is not `exported` in `index.js`, external customers won't be able to use it. I can add a comment to clarify usage.

File: aws-distro-opentelemetry-node-autoinstrumentation/test/sampler/aws-xray-remote-sampler.test.ts (hunk 1 above)
Comment (pxaws) [reply aws-distro-opentelemetry-node-autoinstrumentation/test/sampler/aws-xray-remote-sampler.test.ts:245]: I now understand the difference between the two test cases. The comments above _AwsXRayRemoteSampler helps!

**Review by pxaws — APPROVED** [review]

**Pull request merged**